### DISCUSSION TITLE
Performance between basic and enterprise

### DISCUSSION THREAD

**Discussion by bahientn9x** [emqx/emqx#6051] [Q&A] [answered]
Hi.
I have some questions.
Is there a performance difference between basic and enterprise on the
same server configuration ? About the number of device connections
supported, the total number of messages sent and received per second,
regardless of message size.

**Comment by zmstone** [comment, marked as answer]
Hi @bahientn9x 
Enterprise uses the opensource core. (in fact, opensource is just a sub-tree of the enterprise git repo)
For enterprise edition, there are some configurations tuned for large-size clustering performance.

When it comes to supported connections and message rate.
Both opensource and enterprise editions have no limit on anything (in the sense that you are free to tune all the configurations as you wish).
However the enterprise edition's pricing model is based on supported total number of connections, in that sense there is indeed a total connection limit there.

**Reply by bahientn9x** [reply]
Thanks for answer.
Can i configure on opensource to get same large-size clustering performance as on enterprise and do you know how many connections are limited on opensoure ?
My current total number of connections is 100,000 and growing.
I am planning to use 16 core 32gb ram for 1,000,000 connections with 2 nodes using opensource. Is it imposible ?

**Reply by zmstone** [reply]
It could be possible.
We have reached 2 million connections (without a lot of traffic though) in a single node in similar hardware profile.
However, the total capacity varies a lot  on different traffic patterns,
that's why we almost always recommend our users and customers to run benchmark tests to be sure.

You can use emqtt-bench (docker image: https://hub.docker.com/r/emqx/emqtt-bench) for the test.

**Reply by bahientn9x** [reply]
Thanks a lot.

**Comment by shurrman** [comment]
@zmstone 
could you please be more specific on this:

> enterprise edition's pricing model is based on supported total number of connections, in that sense there is indeed a total connection limit there.

I am trying to test the broker, and unable to have more than 250k connections. 

What are the limits?

**Reply by zmstone** [reply]
Hi @shurrman 
Sorry for the late reply. Have you managed to reach more connections?
If not, we’ll have to get more information to help you troubleshoot.
System/hardware profile?
Emqx logs ? client logs? Error codes?
